### PR TITLE
Lesson 5-7 p1: Character device

### PR DESCRIPTION
**Pull request by a-vodka**
Implementation of basic char dev

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+obj-m += chrdev.o
+
+all:
+	make -C /home/alex/Downloads/linux-5.0.7 M=$(PWD) modules
+
+clean:
+	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile]: Need fix PATH

 There are no hard coded paths! You make out the variable, set the default value, give the opportunity to change the variable

Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile]: Please, add empty line at the end of file

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+module_init(chrdev_init);
+module_exit(chrdev_exit);
+
+
+MODULE_AUTHOR("Oleksii.Vodka <oleksii.vodka@gmail.com>");
+MODULE_DESCRIPTION("Character device driver");
+MODULE_LICENSE("GPL");
+MODULE_VERSION("0.1");
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/chrdev.c:156]: Please, add empty line at the end of file

**Review by galkinletter — CHANGES_REQUESTED** [review]
Need fix PATH

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+static int chrdev_init(void)
+{
+	is_open = 0;
+	data_size = 0;
+
+	data_buffer = kzalloc(buffer_size * sizeof(*data_buffer), GFP_KERNEL);
+	if (!data_buffer) {
+		pr_info("chrdev: register error\n");
+        return -ENOMEM;
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/chrdev.c]: Plz fix indents in the lines 105-108.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Fixing indents are required.

File: 02-KernelModules_CharacterDevice/chrdev.c (hunk 2 above)
Comment (a-vodka) [reply 02-KernelModules_CharacterDevice/chrdev.c:156]: I have added empty line. Github doesn't show it.

**Comment by oleg-h2o** [comment]
Accepted.

**Pull request merged**